pkgs/http/lib/src/browser_client.dart

```diff
@@ -158,6 +180,14 @@ Stream<List<int>> _readBody(BaseRequest request, Response response) async* {
       }
       yield (chunk.value! as JSUint8Array).toDart;
     }
+  } on DOMException catch (e, st) {
+    isError = true;
+
+    if (e.name == 'AbortError') {
+      Error.throwWithStackTrace(const AbortedRequest(), st);
```

Comment:
If `AbortedRequest` becomes a `ClientException` then you could move this logic to `_rethrowAsClientException`.